VoiceAttackResponder/VoiceAttackVariables.cs

```diff
@@ -220,149 +234,190 @@ public static void setJsonValues(ref dynamic vaProxy, string prefix, dynamic jso
-            {
-                Logging.Error("Failed to set current status", ex);
-            }
+                ShipMonitor shipMonitor = ((ShipMonitor)EDDI.Instance.ObtainMonitor("Ship monitor"));
 
-            try
-            {
-                setStarSystemValues(EDDI.Instance.LastStarSystem, "Last system", ref vaProxy);
-            }
-            catch (Exception ex)
-            {
-                Logging.Error("Failed to set last system", ex);
-            }
+                try
+                {
+                    if (shipMonitor?.GetCurrentShip() != Ship)
+                    {
+                        setShipValues(shipMonitor?.GetCurrentShip(), "Ship", ref vaProxy);
+                        Ship = shipMonitor.GetCurrentShip();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Logging.Error("Failed to set current ship values", ex);
+                }
 
-            try
-            {
-                setStarSystemValues(EDDI.Instance.HomeStarSystem, "Home system", ref vaProxy);
+                try
+                {
+                    if (shipMonitor?.shipyard != Shipyard)
+                    {
+                        List<Ship> shipyard = new List<Ship>(shipMonitor?.shipyard);
+                        if (shipyard != null)
+                        {
+                            int currentStoredShip = 1;
+                            foreach (Ship StoredShip in shipyard)
+                            {
+                                setShipValues(StoredShip, "Stored ship " + currentStoredShip, ref vaProxy);
+                                currentStoredShip++;
+                            }
+                            vaProxy.SetInt("Stored ship entries", shipMonitor?.shipyard.Count);
+                        }
+                        Shipyard = shipMonitor.shipyard;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Logging.Error("Failed to set shipyard", ex);
+                }
             }
             catch (Exception ex)
             {
-                Logging.Error("Failed to set home system", ex);
+                Logging.Error("Failed to obtain ship monitor & set VoiceAttack values", ex);
             }
 
             try
             {
-                setSpeechValues(ref vaProxy);
-            }
-            catch (Exception ex)
-            {
-                Logging.Error("Failed to set initial speech service variables", ex);
-            }
+                if (EDDI.Instance.HomeStarSystem != HomeStarSystem)
+                {
+                    setStarSystemValues(EDDI.Instance.HomeStarSystem, "Home system", ref vaProxy);
+                    HomeStarSystem = EDDI.Instance.HomeStarSystem;
 
-            try
-            {
-                setDictionaryValues(EDDI.Instance.State, "state", ref vaProxy);
+                    // Backwards-compatibility with 1.x
```

Comment:
Recommend expanding on this comment for the benefit of our future selves. As someone not familiar with this code my first questions would be 

1. 1.x of what?
2. How do I know that we still need this?
3. How will I know when this can be dropped?

Reply:
Honestly, I'm just preserving what was already there. I have half a mind to just drop it.
And I'm pretty sure that it's a reference to EDDI 1.x.